src/test/regress/expected/multi_task_assignment_policy_0.out

```diff
@@ -176,3 +176,84 @@ RESET client_min_messages;
 DEBUG:  StartTransactionCommand
 DEBUG:  ProcessUtility
 COMMIT;
+-- Check how task_assignment_policy impact planning decisions for reference tables
+-- We rely on the node:ports displayed in the explain plan
+RESET citus.explain_distributed_queries;
+CREATE TABLE task_assignment_reference_table (test_id  integer);
+SELECT create_reference_table('task_assignment_reference_table');
+ create_reference_table 
+------------------------
+ 
+(1 row)
+
+SET citus.task_assignment_policy TO 'greedy';
+EXPLAIN (COSTS FALSE) SELECT * FROM task_assignment_reference_table;
+                                           QUERY PLAN                                           
+------------------------------------------------------------------------------------------------
+ Custom Scan (Citus Router)
+   Task Count: 1
+   Tasks Shown: All
+   ->  Task
+         Node: host=localhost port=57638 dbname=regression
+         ->  Seq Scan on task_assignment_reference_table_880000 task_assignment_reference_table
+(6 rows)
+
+EXPLAIN (COSTS FALSE) SELECT * FROM task_assignment_reference_table;
+                                           QUERY PLAN                                           
+------------------------------------------------------------------------------------------------
+ Custom Scan (Citus Router)
+   Task Count: 1
+   Tasks Shown: All
+   ->  Task
+         Node: host=localhost port=57638 dbname=regression
+         ->  Seq Scan on task_assignment_reference_table_880000 task_assignment_reference_table
+(6 rows)
+
+SET citus.task_assignment_policy TO 'first-replica';
+EXPLAIN (COSTS FALSE) SELECT * FROM task_assignment_reference_table;
+                                           QUERY PLAN                                           
+------------------------------------------------------------------------------------------------
+ Custom Scan (Citus Router)
+   Task Count: 1
+   Tasks Shown: All
+   ->  Task
+         Node: host=localhost port=57638 dbname=regression
+         ->  Seq Scan on task_assignment_reference_table_880000 task_assignment_reference_table
+(6 rows)
+
+EXPLAIN (COSTS FALSE) SELECT * FROM task_assignment_reference_table;
+                                           QUERY PLAN                                           
+------------------------------------------------------------------------------------------------
+ Custom Scan (Citus Router)
+   Task Count: 1
+   Tasks Shown: All
+   ->  Task
+         Node: host=localhost port=57638 dbname=regression
+         ->  Seq Scan on task_assignment_reference_table_880000 task_assignment_reference_table
+(6 rows)
+
+SET citus.task_assignment_policy TO 'round-robin';
+EXPLAIN (COSTS FALSE) SELECT * FROM task_assignment_reference_table;
```

Comment:
Is this deterministic ? If we end up adding any new query above, we would have to get here and fix it. It may get worse if we have parallel tasks doing this.

Reply:
you are right, it could be undeterministic if there is an odd amount of calls to `UniqueJobId` in between which can be cause by queries, however the tests above in this file suffer the same.

I have changed the assertion to be more inline with the tests above.